Lib/test/test_typing.py

```diff
@@ -1785,6 +1789,7 @@ def test_extended_generic_rules_repr(self):
     def test_generic_forward_ref(self):
         def foobar(x: List[List['CC']]): ...
         def foobar2(x: list[list[ForwardRef('CC')]]): ...
+        def foobar3(x: list[ForwardRef('CC')] | int): ...
```

Comment:
Please nest a union inside the ForwardRef too to test the `_eval_type` recursion is correct.

```suggestion
        def foobar3(x: list[ForwardRef('CC | int')] | int): ...
```
The assertion below will need updating as well.